code/game/objects/structures.dm

```diff
@@ -119,9 +119,10 @@
 			return
 		if(user.loc == loc)
 			var/turf/target = get_step(T, dir)
-			if(target.density) //Turf is dense, not gonna work
-				to_chat(user, "<span class='warning'>You cannot leap this way.</span>")
-				return
+			for(var/atom/A in target.contents)
+   				if(A.density)
+					to_chat(user, "<span class='warning'>You cannot leap this way.</span>")
+					return
```

Comment:
Don't remove turf density check or players will "oorah" through wall turfs. Plus this overrides the control below `for(var/atom/movable/A in target)`, which also checks if the structure can be effectively climbed (I.E. window frame, tables).
On the aforementioned control, change the `var/` from `atom/movable/` to `obj/`, and find another way to fix the edge cases.